apps/aecore/src/aec_hard_forks.erl

```diff
@@ -1,79 +1,135 @@
+    #{ ?ROMA_PROTOCOL_VSN     => 0
+     %%, ?MINERVA_PROTOCOL_VSN  => Not yet decided
+     };
+protocols_from_network_id(<<"local_roma_testnet">>) ->
+    #{ ?ROMA_PROTOCOL_VSN     => 0
+     %%, ?MINERVA_PROTOCOL_VSN  => Excluded for testing old protocol
+     };
+protocols_from_network_id(_ID) ->
+    case aeu_env:user_map([<<"chain">>, <<"hard_forks">>]) of
+        undefined ->
+            #{ ?ROMA_PROTOCOL_VSN     => 0
+             , ?MINERVA_PROTOCOL_VSN  => 1
+             };
+        {ok, M} ->
+            maps:from_list(
+              lists:map(
+                fun({K, V}) -> {binary_to_integer(K), V} end,
+                maps:to_list(M)))
+    end.
+
+
+%% Exported for tests
+check_protocol_version_validity(Version, Height, Protocols) ->
+    case maps:is_key(Version, Protocols) of
+        false ->
+            {error, unknown_protocol_version};
+        true ->
+            case protocol_effective_at_height(Height, Protocols) of
+                Version -> ok;
+                Other   -> {error, {protocol_version_mismatch, Other}}
+            end
+    end.
+
+%% Exported for tests
+sorted_protocol_versions() ->
+    lists:sort(maps:keys(protocols())).
+
+protocols_sorted_by_version(Ps) ->
+    lists:keysort(1, maps:to_list(Ps)).
 
--spec protocol_effective_at_height(aec_blocks:height(), aec_governance:protocols()) ->
-                                          version().
 protocol_effective_at_height(H, Protocols) ->
     assert_height(H),
     SortedProtocols = protocols_sorted_by_version(Protocols),
     %% Find the last protocol version effective before or at the
     %% specified height: that is the one effective at the specified
     %% height.  This assumes that the height is strictly increasing
     %% with the version so also assert that for the sake of clarity.
-    Protocols = protocols(Protocols), %% Height is increasing.
     ProtocolsEffectiveSinceBeforeOrAtHeight = [_|_] =
         lists:takewhile(fun({_, HH}) -> H >= HH end, SortedProtocols),
     {V, _} = lists:last(ProtocolsEffectiveSinceBeforeOrAtHeight),
     V.
 
-%%%===================================================================
-%%% Internal functions
-%%%===================================================================
-
-assert_height(H) ->
-    case is_integer(H) andalso H >= aec_block_genesis:height() of
-        true  -> ok;
-        false -> error({illegal_height, H})
-    end.
-
-sorted_versions() ->
+assert_protocols(M) ->
     GenesisVersion = aec_block_genesis:version(),
-    Vs = [GenesisVersion | _] = aec_governance:sorted_protocol_versions(),
+    Vs = [GenesisVersion | _] = sorted_protocol_versions(),
     %% Assert versions are sorted.
     Vs = lists:sort(Vs),
     %% Assert versions are distinct.
     [] = lists:sort(Vs) -- lists:usort(Vs),
-    Vs.
+    {[], _} = {maps:keys(M) -- Vs, check_no_extra_protocol_versions},
+    {[], _} = {Vs -- maps:keys(M), check_no_missing_protocol_versions},
```

Comment:
Considering how assert_protocols is called (one place), this is a tautology.

Reply:
This seems to be the case from this change https://github.com/aeternity/aeternity/commit/84e1f517a753d10f62a3cbff9023925e099ac1c5#diff-312e5e881622101a1720cef694345ee9R71